docs/content/direct/multi-wec-aggregated-status.md

```diff
@@ -0,0 +1,197 @@
+
+This ensures required status fields are populated correctly but not limited to only that. We provide best effort to aggregate other status fields as much as possible.
+
+Key objectives:  
+- Maintain backward compatibility when `wantMultiWECReportedState` is false, preserving existing behavior.  
+- When `wantMultiWECReportedState` is enabled, provide fallback to the singleton mechanism if only one WEC is bound.  
+- Aggregate per-WEC `WorkStatus` objects and update the `.status` field in WDS workload objects with combined readiness and health status.
+
+---
+
+## Solution Approach
+
+### 1. Extend BindingPolicy CRD
+
+The configuration bit `wantMultiWECReportedState` is added under each `DownsyncModulation` struct (not directly under `BindingPolicy`), analogously to how [singleton state return ](./combined-status.md#special-case-for-1-wec) is requested.
+
+Example:
+
+```yaml
+apiVersion: policies.kubestellar.io/v1alpha1
+kind: BindingPolicy
+metadata:
+  name: multiwec-nginx
+spec:
+  clusterSelectors:
+  - matchLabels: {"location": "edge"}
+  downsync:
+  - objectSelectors:
+    - matchLabels: {"app.kubernetes.io/name": "nginx-multi"}
+    wantMultiWECReportedState: true
+```
+
+This example follows the [singleton status example](https://docs.kubestellar.io/release-0.28.0/direct/example-scenarios/#scenario-4-singleton-status) but replaces `wantSingletonReportedState` with `wantMultiWECReportedState`.
+
+---
+
+### 2. Introduce `handleMultiWECReportedState()` Function (Planned)
+
+A new controller function, `handleMultiWECReportedState()`, will implement multi-WEC aggregated status reporting using per-kind aggregation logic for common workload kinds such as Deployment, StatefulSet, DaemonSet, ReplicaSet, Job, CronJob, and Pod.
+
+Both `BindingPolicy` and `Binding` inherit the `DownsyncModulation` extension where this flag resides.
+
+---
+
+### 3. Status Aggregation Rules
+
+The aggregation logic depends on the **workload object kind**.
+
+#### Case 1 – Known Workload Kinds
+
+For workload kinds recognized by ArgoCD, the controller applies predetermined field aggregation rules consistent with ArgoCD’s native health evaluation. These include:
+
+##### **Deployment**  
+  - `replicas`, `updatedReplicas`, `availableReplicas`, `readyReplicas`: use `min()` across clusters to reflect the least ready state.  
+  - `conditions`: apply three-value logic per condition type:  
+    - `False` if any cluster reports `False`.  
+    - `True` if all clusters report `True`.  
+    - Otherwise, `Unknown`.  
+  - `observedGeneration`: use the minimum observed generation.  
+  - `unavailableReplicas`: use `max()` to represent the worst state.
+
+##### **StatefulSet**  
+  - `replicas`, `readyReplicas`, `currentReplicas`, `updatedReplicas`: use `min()`.  
+  - `conditions`: three-value logic as above.
+
+##### **DaemonSet**  
+  - `currentNumberScheduled`, `desiredNumberScheduled`, `numberAvailable`, `numberReady`: use `min()`.  
```

Comment:
What about `updatedNumberScheduled`? That seems like the best match to https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#deployment-replicaset-statefulset-daemonset to me.